Windows build doc update 2025 11 09

lioobayoyo opened the pull request:
* Updated BUILDING.md instructions for Windows as per 2025-11-09 (VS version, QT links)
* Add a hint in readme for quick memory map manual update

lioobayoyo committed:
Windows build instructions (VS 2022 and links for latest qt installer)

lioobayoyo committed:
hint for memory maps manual update in readme

lioobayoyo committed:
typo / phrasing

lioobayoyo commented on BUILDING.md:
```diff
@@ -13,27 +13,32 @@ For building the manual, see the README in the [Manual repository](https://githu
 
-Install "Tools for Visual Studio 2019" and
-select these additional options:
+Make sure "Development Tools for .NET" and "MSBuild" are included in the installtion options.
```
```suggestion
Make sure "Development Tools for .NET" and "MSBuild" are included in the installation options.
```

cvuchener commented:
I wanted to try with vcpkg. It is the simplest way to get the latest Qt 5 version. But it builds Qt from source so it takes a very long time. Also I could not get windeployqt to work with it. So I am abandoning this for now.

cvuchener commented on BUILDING.md:
```diff
@@ -13,27 +13,32 @@ For building the manual, see the README in the [Manual repository](https://githu
+
+You can either :
+* use the [official installer](https://doc.qt.io/qt-6/get-and-install-qt.html).
+* or use direct download links from the archive at https://download.qt.io/new_archive/qt, for instance [Qt 5.14.2 here](https://download.qt.io/new_archive/qt/5.14/5.14.2/qt-opensource-windows-x86-5.14.2.exe)
```
This a link for the wrong version. I think Qt stopped providing offline installer at some point. The online installer is the only option (for binary releases but building from source is long and complex).

lioobayoyo replied:
This isn't the last version 5.x, however it seems any version 5.9 or higher works, doesn't it ? 
In my case, I was able to use the releases in this folder in order to build, it was much easier than using the official installer.

cvuchener commented on README.rst:
```diff
@@ -28,6 +28,16 @@ Dwarf-Therapist requires a C++ compiler (with C++14 support), cmake (3.1.0 or ne
+
+To update manually you can either:
+
+- build from source code yourself, as per the section above;
```
Building from source for updating memory layouts is useless and complex. I would not recommend that in the readme.

lioobayoyo replied:
removed.

cvuchener commented on README.rst:
```diff
@@ -28,6 +28,16 @@ Dwarf-Therapist requires a C++ compiler (with C++14 support), cmake (3.1.0 or ne
+To update manually you can either:
+
+- build from source code yourself, as per the section above;
+- or grab the memory maps corresponding to your installed DF version from the [/share/data](https://github.com/lioobayoyo/Dwarf-Therapist/tree/master/share), and copy them into the corresponding data folder of your Dwarf Therapist installation.
```
You are linking your own repository there. I think a relative link should work there. Or use a link to the main repository.

lioobayoyo replied:
oops, my bad, fixed

cvuchener commented:
The section about manual updates may be the place to add instructions for the DFHack script for generating memory layouts. See #325.

If the data directory is accessed through the *File* menu. It should contains [this readme](https://github.com/Dwarf-Therapist/Dwarf-Therapist/blob/master/share/README.md).

cvuchener committed:
Update BUILDING.md

Co-authored-by: lioobayoyo <lioobayoyo@users.noreply.github.com>

lioobayoyo committed:
address review comments for readme - link fix and rephrasing

lioobayoyo committed:
Merge branch 'build_doc_update_2025-11-09' of https://github.com/Lioobayoyo/Dwarf-Therapist into build_doc_update_2025-11-09

lioobayoyo commented:
> The section about manual updates may be the place to add instructions for the DFHack script for generating memory layouts. See #325.
> 
> If the data directory is accessed through the _File_ menu. It should contains [this readme](https://github.com/Dwarf-Therapist/Dwarf-Therapist/blob/master/share/README.md).

I'll test it and include new instructions, great suggestion

lioobayoyo committed:
link to dt-memory-layout tool to create memory maps

lioobayoyo commented:
> > The section about manual updates may be the place to add instructions for the DFHack script for generating memory layouts. See #325.

* Added build instructions for df-memory-layout in its repository : https://github.com/cvuchener/dt-memory-layout/pull/1
* Updated link in README.md to link to df-memory-layout

> > If the data directory is accessed through the _File_ menu. It should contains [this readme](https://github.com/Dwarf-Therapist/Dwarf-Therapist/blob/master/share/README.md).

I'm unsure what you mean. Do you want me to copy it somewhere else ?

lioobayoyo commented:
> > The section about manual updates may be the place to add instructions for the DFHack script for generating memory layouts. See #325.
> > If the data directory is accessed through the _File_ menu. It should contains [this readme](https://github.com/Dwarf-Therapist/Dwarf-Therapist/blob/master/share/README.md).
> 
> I'll test it and include new instructions, great suggestion

Built and tested. However, to use df-memory-layout, it's unclear what the "version name" should be in command line.

I tried using the same version names as in the main XML tags in df-structures (`"v0.53.03 win64 STEAM"` ) or like in  .ini filenames from dwarf therapist ( (`"v0.53.05-steam_win64"`), I always get errors:

```
PS D:\games\df\dt-memory-layout\out\build\x64-Debug> .\dt-memory-layout.exe D:\games\df\df-structures "v0.53.03 win64 STEAM" D:\games\df\dt-memory-layout\ini\memory_layout.xml > v0.53.05-steam_win64.ini
Cannot resolve creature_handler.hist_fig_to_creature_map member  reference to key-type
Cannot resolve creature_handler.hist_fig_to_creature_map member  reference to value-type
Cannot resolve filter_entry member filtered_set reference to stl-unordered-set
Cannot resolve ....  (many lines similar)
Could not load structures: Failed to parse structures xml
```
(and got no file output).

Or it could be related to the [df-structures](https://github.com/DFHack/df-structures) not being up-to-date ? (I see pipeline failures for DF latest version 53.05).

lioobayoyo commented:
> Built and tested. However, to use df-memory-layout, it's unclear what the "version name" should be in command line.
> 
> I tried using the same version names as in the main XML tags in df-structures (`"v0.53.03 win64 STEAM"` ) or like in .ini filenames from dwarf therapist ( (`"v0.53.05-steam_win64"`), I always get errors:
> 
> ```
>....
> Cannot resolve ....  (many lines similar)
> Could not load structures: Failed to parse structures xml
> ```
> 
> (and got no file output).

Added an [issue](https://github.com/cvuchener/dt-memory-layout/issues/2)
